sympy/solvers/tests/test_solvers.py

```diff
@@ -27,6 +28,12 @@
 def NS(e, n=15, **options):
     return sstr(sympify(e).evalf(n, **options), full_prec=True)
 
+def test_floor():
+    assert solve(floor(x)) == [0, 1]
```

Comment:
`1` is not a solution to `floor(x) == 0`.

Reply:
Yes, I will use intervals with right open to give lower and upper limits wherever required.

Reply:
What about when the solution requires multiple intervals? For example, `floor(x^2) = 1`.

Reply:
For those the current implementation of the function gives incorrect results. I will try to fix them as much as possible.

Reply:
Since I have revived the PR, I have noticed that certain parts of the code needs to be changed like: ```math.ceil``` has to be removed from the code and the equation that you are suggesting gives correct results. I will make other changes as well and am planning to return a set of Intervals which will mostly be right open or left open wherever required.

Reply:
@ethankward  In the latest commit, the ```solve_floor``` function solves ```floor(x^2) = 1``` properly and returns answers in terms of Intervals.